src/pages/SelectCompany.tsx

```diff
@@ -105,7 +75,19 @@ const SelectCompany = ({ token }: { token: string }) => {
                   className="px-14 mr-4"
                   color="primary"
                   tag="button"
-                  onClick={saveSelectedCompany}
+                  onClick={() => {
+                    if (
+                      authentication.currentSession?.type === "user" &&
+                      selectedCompany
+                    ) {
+                      authentication.setCurrentSession({
+                        type: "user",
+                        userFiscalCode:
+                          authentication.currentSession.userFiscalCode,
+                        merchantFiscalCode: selectedCompany
+                      });
+                    }
+                  }}
```

Comment:
Extract this logic inside a function (like `onClick` or `handleAuthentication`) and pass it here
```suggestion
      onClick={onClick} // or handleAuthentication
```

Reply:
adressed in fc6721a9357ca484da1a0dd4051ab66d54959dfd